### PR TITLE
Add RSS for tags

### PR DESCRIPTION
**Pull request by j0k3r**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Documentation | not needed
| Translation   | no
| Fixed tickets | https://github.com/wallabag/wallabag/issues/3032 & https://github.com/wallabag/wallabag/issues/2769
| License       | MIT

Adding RSS for tags.
Url are `https://wallabag.instance/admin/43kIG075GUoLZiU/tags/foot.xml`

Also, I'm wondering where should we display RSS link for tag.
- put a sample url in the config page next to other RSS link (like "for tag please use that pattern `/username/token/tags/tag-slug.xml`")
- put a link on the tag page

### CODE REVIEW

**Comment by nicosomb** [comment]
👍  for `put a link on the tag page`, or near the tag name on full tags page.

**Comment by tcitworld** [comment]
Can't choose between the two, will have a look at how it feels.

File: src/Wallabag/CoreBundle/Controller/ExportController.php [hunk 1]
```diff
@@ -57,16 +56,17 @@ public function downloadEntriesAction(Request $request, $format, $category)
...
+        $epository = $this->get('wallabag_core.entry_repository');
 
         if ($category == 'tag_entries') {
-            $tag = $this->getDoctrine()->getRepository('WallabagCoreBundle:Tag')->findOneBySlug($request->query->get('tag'));
+            $tag = $this->get('wallabag_core.tag_repository')->findOneBySlug($request->query->get('tag'));
 
-            $entries = $this->getDoctrine()
-                ->getRepository('WallabagCoreBundle:Entry')
-                ->findAllByTagId($this->getUser()->getId(), $tag->getId());
+            $entries = $epository->findAllByTagId(
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Controller/ExportController.php]: typo

File: src/Wallabag/CoreBundle/Controller/ExportController.php [hunk 2]
```diff
@@ -57,16 +56,17 @@ public function downloadEntriesAction(Request $request, $format, $category)
...
+                $this->getUser()->getId(),
+                $tag->getId()
+            );
         } else {
-            $entries = $this->getDoctrine()
-                ->getRepository('WallabagCoreBundle:Entry')
+            $entries = $epository
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Controller/ExportController.php]: same

**Comment by j0k3r** [comment]
Feeds icon screen:

image

Shouldn't we add an icon somewhere in a tag page (I mean not only on the tag**s** page)

**Comment by tcitworld** [comment]
Ideally, yes. And possibly a <link> tag to it in the headers.

**Comment by jamietanna** [comment]
How will the URLs be generated? Will it be possible to regenerate them ie if you want to stop someone being able to access the feed? Also, will it be possible to have multiple URLs for the same feed?

**Comment by j0k3r** [comment]
Like I said in the PR description:

> https://wallabag.instance/USERNAME/RSS_TOKEN/tags/TAG_SLUG.xml

If someone found your rss token and then can discover your feed, you just need have to regenerate your RSS token from your config panel and update your RSS reader with the new feed url.

What do you mean by multiple URLs for the same feed? Please describe what you need.

**Comment by jamietanna** [comment]
Ah sorry I missed that. That makes sense.

> What do you mean by multiple URLs for the same feed?

Say I want to share the `work` tag with my work email, and my personal phone (via an RSS reader). But I want the chance to revoke the access to the work URL at any point. Having multiple, separate URLs would help this.

**Comment by j0k3r** [comment]
You mean multiple tokens for each url I think.

**Comment by jamietanna** [comment]
Yes, exactly

**Comment by j0k3r** [comment]
@jamietanna there is already an issue about that https://github.com/wallabag/wallabag/issues/626.

**Comment by tcitworld** [comment]
- Added « all entries » RSS feed
- Added rss button on all views
- Added tags rss on baggy
- Fixed some translation issues

**Comment by j0k3r** [comment]
@tcitworld great, an other fixed tickets.
I've tested it on both baggy & material and it looks good to me :+1:

File: src/Wallabag/CoreBundle/Resources/views/themes/material/Entry/entries.html.twig [hunk 3]
```diff
@@ -1,9 +1,16 @@
 {% extends "WallabagCoreBundle::layout.html.twig" %}
 
+{% block head %}
```
Comment (tcitworld) [inline src/Wallabag/CoreBundle/Resources/views/themes/material/Entry/entries.html.twig]: This part is missing on baggy.

Comment (j0k3r) [inline src/Wallabag/CoreBundle/Resources/views/themes/material/Entry/entries.html.twig]: Added

**Pull request merged**